### DISCUSSION TITLE
compile openvaf for Ubuntu and Apple

### DISCUSSION THREAD

**Discussion by fidel2030** [pascalkuthe/OpenVAF#118] [General]
When compiling in Ubuntu this function gets called fn gen_msvcrt_importlib(sh: &Shell, arch: &str, target: &str, check: bool)
It appends  target and calls command: 
"clang-cl /c /Zl /GS- /clang:--target={target}-pc-windows-msvc /clang:-o{ucrt_obj} {ucrt_src}"

Is this for Windows compiling? How to change it to Ubuntu, on an Ubuntu computer, and to  x86_64-apple-darwin for Apple?

**Comment by mi-and-core** [comment]
tested on Ubuntu 22.04 and Ubuntu 24.04

```
sudo apt install -y g++ cargo
mkdir -p temp && cd temp

wget https://openva.fra1.cdn.digitaloceanspaces.com/llvm-15.0.7-x86_64-unknown-linux-gnu-FULL.tar.zst
zstd -d -c --long=31 llvm-15.0.7-x86_64-unknown-linux-gnu-FULL.tar.zst | tar -xf -

LLVM_CONFIG=${PWD}/LLVM/bin/llvm-config
PATH=${PWD}/LLVM/bin:$PATH

git clone https://github.com/pascalkuthe/OpenVAF.git && cd OpenVAF
git checkout "a9697ae7780518f021f9f64e819b3a57033bd39f"

cargo build --release --bin openvaf

```
